js/utils/localServer.js

```diff
@@ -0,0 +1,139 @@
+import _ from 'underscore';
+import { EOL, platform } from 'os';
+import { Events } from 'backbone';
+import fs from 'fs';
+import childProcess from 'child_process';
+
+export default class LocalServer {
+  constructor(options) {
+    if (!options.serverPath) {
+      throw new Error('Please provide a server path.');
+    }
+
+    if (!options.serverFilename) {
+      throw new Error('Please provide a server filename.');
+    }
+
+    if (!options.errorLogPath) {
+      throw new Error('Please provide an error log path.');
+    }
+
+    _.extend(this, Events);
+    this.serverPath = options.serverPath;
+    this.serverFilename = options.serverFilename;
+    this.errorLogPath = options.errorLogPath;
+    this._isRunning = false;
+    this._debugLog = '';
+    this.startAfterStop = () => this.start();
+  }
+
+  get isRunning() {
+    return this._isRunning;
+  }
+
+  start() {
+    if (this.pendingStop) {
+      this.pendingStop.once('close', this.startAfterStop);
+      const debugInfo = 'Attempt to start server while an existing one' +
+        ' is the process of shutting down. Will start after shut down is complete.';
+      this.log(debugInfo);
+      return;
+    }
+
+    if (this.isRunning) return;
+    this._isRunning = true;
+
+    this.log('Starting local server.');
+    console.log('Starting local server.');
+
+    this.serverSubProcess = childProcess.spawn(this.serverPath + this.serverFilename, ['start'], {
```

Comment:
@hoffmabc I haven't looked at that alternate module you suggested which Slack is using. I was too far deep in with this one. I may still very well check it out (particularly if we find we're having issues with this one).